examples/context7-rs/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+            };
+        }
+
+        let encoded_query = encode(trimmed);
+        let endpoint = format!("/v1/search?query={encoded_query}");
+        match make_request(Method::Get, &endpoint, None, get_api_key().as_deref()) {
+            Ok(body) => match parse_search_response(&body) {
+                Ok(response) => response,
+                Err(message) => SearchResponse {
+                    success: false,
+                    results: Vec::new(),
+                    error: Some(message),
+                },
+            },
+            Err(message) => SearchResponse {
+                success: false,
+                results: Vec::new(),
+                error: Some(message),
+            },
+        }
+    }
+
+    fn get_library_docs(
+        context7_compatible_library_id: String,
+        topic: Option<String>,
+        tokens: Option<u32>,
+    ) -> DocsResponse {
+        let trimmed = context7_compatible_library_id.trim();
+        if trimmed.is_empty() {
+            return DocsResponse {
+                success: false,
+                content: None,
+                error: Some("Context7 compatible library ID is required".to_string()),
+            };
+        }
+
+        let library_id = trimmed.trim_start_matches('/').to_string();
+        if library_id.is_empty() {
+            return DocsResponse {
+                success: false,
+                content: None,
+                error: Some("Context7 compatible library ID is required".to_string()),
+            };
+        }
+
+        let token_count = tokens
+            .map(|value| value.max(DEFAULT_MINIMUM_TOKENS))
+            .unwrap_or(DEFAULT_MINIMUM_TOKENS);
+
+        let mut parameters = vec![format!("tokens={token_count}"), "type=txt".to_string()];
+
+        if let Some(topic_value) = topic.as_ref().map(|value| value.trim()).filter(|v| !v.is_empty()) {
+            parameters.push(format!("topic={}", encode(topic_value)));
+        }
+
+        let endpoint = format!("/v1/{}?{}", library_id, parameters.join("&"));
+        match make_request(Method::Get, &endpoint, None, get_api_key().as_deref()) {
+            Ok(body) => {
+                let body_trimmed = body.trim();
+                if body_trimmed.is_empty() || NO_CONTENT_MESSAGES.iter().any(|msg| body_trimmed == *msg) {
+                    DocsResponse {
+                        success: false,
+                        content: None,
+                        error: Some("No documentation available for this library".to_string()),
+                    }
+                } else {
+                    DocsResponse {
+                        success: true,
+                        content: Some(body),
+                        error: None,
+                    }
+                }
+            }
+            Err(message) => DocsResponse {
+                success: false,
+                content: None,
+                error: Some(message),
+            },
+        }
+    }
+}
+
+fn make_request(
```

Comment:
If you don't want to do all of this low level http code, consider importing https://docs.rs/wasmcloud-component/latest/wasmcloud_component/http/index.html. It has all the convenience wrappers around the http request stuff so you can just build requests and read responses using `std::io::Read`